### ISSUE TITLE
A problem in the example for pix2pix

### ISSUE THREAD

**Issue by EngrStudent** [tensorflow/tensorflow#55917]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type
Others

### Current Behavior?

```shell
On this site:
https://github.com/tensorflow/docs/blob/master/site/en/tutorials/generative/pix2pix.ipynb

The output of the generator is tanh, but the image it is compared to is [0,...,1].  
This means half the dynamic range of the activation is wasted.  
This is going to slow down learning.

Where it says this:
last = tf.keras.layers.Conv2DTranspose(OUTPUT_CHANNELS, 4,
                                         strides=2,
                                         padding='same',
                                         kernel_initializer=initializer,
                                         activation='tanh') 

It might be better to say this:
last = tf.keras.layers.Conv2DTranspose(OUTPUT_CHANNELS, 4,
                                         strides=2,
                                         padding='same',
                                         kernel_initializer=initializer,
                                         activation='sigmoid')
```

### Standalone code to reproduce the issue

```shell
#test the discriminator

img_in = inp/255.
print(np.max(img_in))
print(np.min(img_in))

plt.imshow(img_in)
plt.colorbar()
plt.title("Input image Left")
plt.show()

gen_out = tf.keras.backend.squeeze(gen_output,0)
print(np.max(gen_out))
print(np.min(gen_out))
plt.imshow(gen_out)
plt.colorbar()
plt.title("Input image Right")
plt.show()

#put image pair through the discriminator
disc_out = discriminator([inp[tf.newaxis, ...], gen_output], training=False)

#show the discriminator output
# NOTE: the size is MUCH smaller than the input image.
plt.imshow(disc_out[0, ..., -1], vmin=-20, vmax=20, cmap='RdBu_r')
plt.title("Discriminator Output (Note the axes scales)")
plt.colorbar()
```

### Relevant log output

```shell
#this is the actual image range
1.0
0.0

#this is the warning it gives for the need to clip
Clipping input data to the valid range for imshow with RGB data ([0..1] for floats or [0..255] for integers).

#this is the actual range of the generator output
1.0
-1.0

<matplotlib.colorbar.Colorbar at 0x1c286147310>
```
</details>

**Comment by chunduriv** [comment]
@EngrStudent,

Sorry for the late response. I have created code fix internally. Thank you.

**Comment by MarkDaoust** [comment]
I disagree. 

Refer to the `normalize` and `generate_images` functions in the tutorial.
